lib/index.js

```diff
@@ -267,36 +267,58 @@ King = new Class({
     /**
      * Generates a valid url for a kong request, cycling through one of the hosts for each call
      * @method module:kingkong#request
-     * @param {String} resource The type of resource to generate a url for ( api, plugin, consumer ) 
+     * @param {String} resource The type of resource to generate a url for ( api, plugin, consumer )
      * @param {String} [id] the id or name of a specific resource
-     * @return {Promise} A promise that will resolve to the result of the request 
+     * @return {Promise} A promise that will resolve to the result of the request
      **/
-    ,request:function _request( action, resource, id, data ){
-        let result = resource_exp.exec( resource )
-          , method = action.toUpperCase()
-          , body   = clone( data )
-          , error
-          , uri
-          ;
+    ,request:function _request( action, resource, id, data, optionalAcceptableStatusCodes ){
+        let requestOptions;
 
-        if(!result){
-            error = new Error('invalid resource type ' + resource);
-            error.name = 'InvalidResourceError';
-            throw error;
+        //  Sometimes we send entire request options object as first parameter in which case
+        //  we don't need to generate the request options.
+        if (typeof action === 'object') {
+            requestOptions = action;
+            optionalAcceptableStatusCodes = resource;
+        } else {
+            let result = resource_exp.exec( resource )
+              , method = action.toUpperCase()
+              , body   = clone( data )
+              , error
+              , uri
+              ;
+
+            if(!result) {
+                error = new Error('invalid resource type ' + resource);
+                error.name = 'InvalidResourceError';
+                throw error;
+            }
+
+            result = result[1] + 's';
+            uri    = this.url( result, id );
+
+            requestOptions = {
+                method:method
+              , uri: uri
+              , json:true
+              , body: body
+            };
         }
 
-        result = result[1] + 's';
-        uri    = this.url( result, id ); 
-        return request({
-            method:method
-          , uri: uri
-          , json:true
-          , body: body
+        const acceptableStatusCodes = optionalAcceptableStatusCodes || [200, 201]
+
+        return request(requestOptions).then((requestResult) => {
+            if (acceptableStatusCodes.find(statusCode => statusCode === requestResult.statusCode)) {
```

Comment:
Um... No, I think we both made a mistake here. For `in` to work we would have to create an object (`{200: 200, 201: 201`} rather than an array (`[200, 201]`)